packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -227,6 +226,15 @@ const assertRequesterIsDelegateProducer = (
   }
 };
 
+const assertRequesterIsDelegate = (
+  delegateId: TenantId | undefined,
+  authData: AuthData
+): void => {
+  if (authData.organizationId !== delegateId) {
+    throw operationNotAllowed(authData.organizationId);
+  }
+};
```

Comment:
In #1246 I removed this to avoid having a method too generic on delegations, without consumer/producer mention - in `assertRequesterIsDelegateProducer` I did the delegate check together with the rest of the checks in the if without calling `assertRequesterIsDelegate`.
We can also leave it, with no strong opinion, in any case, could you align the two methods `assertRequesterIsDelegateProducer` and `assertRequesterIsDelegateConsumer`? 🙏